tests/snapshots/compile_file_o_all__inline_app.hvm.snap

```diff
@@ -2,7 +2,9 @@
 source: tests/golden_tests.rs
 input_file: tests/golden_tests/compile_file_o_all/inline_app.hvm
 ---
-@Rul3 = (a a)
+@Rul1_$_Rul2 = (a a)
+
+@Rul3 = @Rul1_$_Rul2
 
 @main = a
   & @Rul3 ~ (@Rul3 a)
```

Comment:
Does the inlining not work on the entrypoint?

Reply:
I believe in this test it does not work because of the constructor node or because Rul3 is pointing to another reference

Reply:
I expected @Rul3 to inline to @Rul1_$_Rul2 here. It should be inlining Ref nodes since they have size 0.